### ISSUE TITLE
test command: infinite loop

### ISSUE THREAD

**Issue by marvinhagemeister** [denoland/deno#18565]
Noticed that the `deno test` command runs into an infinite loop and keeps consuming 100% CPU with no end in sight. My rust profiling skills aren't that strong, but from what I can with macOS's Instruments app is that it seems to be stuck in `swc`. Might be a bug there, but I'm not sure how to inspect further.

<img width="620" alt="Screenshot 2023-04-02 at 12 42 37" src="https://user-images.githubusercontent.com/1062408/229348030-9fcc1958-e13a-453e-9085-6d9c068d00d3.png">

<img width="819" alt="Screenshot 2023-04-02 at 12 40 13" src="https://user-images.githubusercontent.com/1062408/229347954-5e50286b-0a0a-41c7-8f32-1876003a8857.png">

Problem occurs with `deno 1.32.3`.

Steps to reproduce:

1. Clone https://github.com/preactjs/preact-render-to-string (`git clone git@github.com:preactjs/preact-render-to-string.git`)
2. Cd into cloned repo `cd preact-render-to-string`
3. Run `deno test`

**Comment by GJZwiers** [comment]
Did a bisect and found this started in 0ee64ad84726f7c91842c6a2dab0cb6a96dead4b

**Comment by marvinhagemeister** [comment]
Can confirm that this is a bug in `swc`. Parsing this source without enabling jsx triggers the issue.

```jsx
const a = (
	<section>
		<a href="/foo">foo</a>
	</section>
);
```

I'll report it on their repo.

**Comment by marvinhagemeister** [comment]
Fix got merged, now waiting for a new `swc` release.

**Comment by bartlomieju** [comment]
@marvinhagemeister usually @dsherret does the SWC update before the minor release as it involves updating multiple `deno_*` crates in a cascading fashion. I believe you will have to update a few more crates that depend on `deno_ast` (like `deno_graph` and `eszip`)

**Comment by nayeemrmn** [comment]
Fixed by #18795.
